### PR TITLE
Karakalpak lang

### PR DESCRIPTION
**Pull request by saylaww**
---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/generator-constants.mjs [hunk 1]
```diff
@@ -88,6 +88,138 @@ export const LOGIN_REGEX_JS = '^[a-zA-Z0-9!$&*+=?^_`{|}~.-]+@[a-zA-Z0-9-]+(?:\\\
...
+  },
+  {
+    name: 'Uzbek (Latin)',
+    dispName: 'O`zbekcha',
+    value: 'uz-Latn-uz',
+    localeId: 'uz-Latn',
+    dayjsLocaleId: 'uz-latn',
+  },
+  { name: 'Vietnamese', dispName: 'Tiếng Việt', value: 'vi' },
+];
```
Comment (mshima) [inline generators/generator-constants.mjs:222]:
```suggestion
```

File: generators/server/templates/src/test/java/package/service/MailServiceIT.java.ejs [hunk 2]
```diff
@@ -225,7 +225,13 @@ class MailServiceIT {
             verify(javaMailSender, atLeastOnce()).send(messageCaptor.capture());
             MimeMessage message = messageCaptor.getValue();
 
+<<<<<<< HEAD
+            System.out.println("Testing language: " + langKey);
+            System.out.println("getJavaLocale: " + getJavaLocale(langKey));
+            String propertyFilePath = "i18n/messages_" + getJavaLocale(langKey) + ".properties";
+=======
```
Comment (mshima) [inline generators/server/templates/src/test/java/package/service/MailServiceIT.java.ejs:232]:
```suggestion
```

File: generators/languages/support/languages.mts [hunk 3]
```diff
@@ -247,6 +247,13 @@ const partialLanguages: PartialLanguage[] = [
     dayjsLocale: 'uz-latn',
   },
   { name: 'Vietnamese', displayName: 'Tiếng Việt', languageTag: 'vi' },
+  {
+    name: 'Karakalpak (Latin)',
+    displayName: 'Qaraqalpaqsha',
+    languageTag: 'kr-Latn-kr',
+    angularLocale: 'kr-Latn',
+    dayjsLocale: 'kr-latn',
```
Comment (mshima) [inline generators/languages/support/languages.mts:255]: angular doesn't support `kr-Latn`
```suggestion
    dayjsLocale: 'en',
```

Comment (mshima) [inline generators/languages/support/languages.mts:255]: dayjs doesn't support `kr-Latn`
```suggestion
    dayjsLocale: 'en',
```

File: generators/languages/support/languages.mts [hunk 4]
```diff
@@ -247,6 +247,13 @@ const partialLanguages: PartialLanguage[] = [
     dayjsLocale: 'uz-latn',
   },
   { name: 'Vietnamese', displayName: 'Tiếng Việt', languageTag: 'vi' },
+  {
+    name: 'Karakalpak (Latin)',
+    displayName: 'Qaraqalpaqsha',
+    languageTag: 'kr-Latn-kr',
+    angularLocale: 'kr-Latn',
```
Comment (mshima) [inline generators/languages/support/languages.mts:254]: angular doesn't support kr-Latn
```suggestion
    angularLocale: 'en',
```

**Comment by saylaww** [comment]
Are all the bugs over? Now the Karakalpak language has been added to the jhipster generator?

**Review by mraible — APPROVED** [review]
LGTM! I'll let someone with more i18n experience merge it.

**Comment by DanielFran** [comment]
Thanks @saylaww for the contribution

**Comment by saylaww** [comment]
When it's all over, Why isn't the Karakalpak language included in the jhipster generator?

**Comment by DanielFran** [comment]
@saylaww it has been merged together main, but we did not release a new version of the generator that will include it

**Comment by saylaww** [comment]
> @saylaww it has been merged together main, but we did not release a new version of the generator that will include it

When will the new generator come out?

**Comment by mraible** [comment]
You don't need to wait for a release to use it. However, waiting for the release will likely guarantee that everything works. You can use the `main` branch using the instructions in this video.

https://www.youtube.com/watch?v=0UFWJ3JGZa0

For a shortcut, this Gist has the commands:

https://gist.github.com/mraible/1e9b8b755ad6ea7c3dcd86fc860a6d56

**Pull request merged**